core/dbt/contracts/graph/model_config.py

```diff
@@ -438,6 +438,34 @@ class SeedConfig(NodeConfig):
 class TestConfig(NodeConfig):
     materialized: str = 'test'
     severity: Severity = Severity('ERROR')
+    where: Optional[str] = None
+    limit: Optional[int] = None
+    fail_calc: str = "count(*)"
+    warn_if: str = "!= 0"
+    error_if: str = "!= 0"
+
+    @classmethod
+    def same_contents(
```

Comment:
Ah! I see now that this is the proper place for it. Nice work, this logic is much clearer.